src/api/procedures/setAssetStats.ts

```diff
@@ -1,29 +1,98 @@
-import { PolymeshPrimitivesStatisticsStatType } from '@polkadot/types/lookup';
+import {
+  PolymeshPrimitivesStatisticsStatType,
+  PolymeshPrimitivesStatisticsStatUpdate,
+} from '@polkadot/types/lookup';
+import { BTreeSet } from '@polkadot/types-codec';
 
-import { Procedure } from '~/internal';
-import { FungibleAsset, SetTransferRestrictionStatParams, StatType, TxTags } from '~/types';
-import { BatchTransactionSpec, ProcedureAuthorization, TxWithArgs } from '~/types/internal';
+import { Context, PolymeshError, Procedure } from '~/internal';
+import {
+  AddBalanceStatParams,
+  AddClaimBalanceStatParams,
+  AddClaimCountStatParams,
+  AddCountStatParams,
+  ErrorCode,
+  FungibleAsset,
+  SetTransferRestrictionStatParams,
+  StatType,
+  TxTag,
+  TxTags,
+} from '~/types';
+import { BatchTransactionSpec, ProcedureAuthorization } from '~/types/internal';
 import {
   assetToMeshAssetId,
+  balanceStatInputToStatUpdates,
+  claimBalanceStatInputToStatUpdates,
   claimCountStatInputToStatUpdates,
   claimIssuerToMeshClaimIssuer,
   countStatInputToStatUpdates,
+  inputStatTypeToMeshStatType,
   statisticsOpTypeToStatType,
   statisticStatTypesToBtreeStatType,
   statTypeToStatOpType,
 } from '~/utils/conversion';
-import { checkTxType } from '~/utils/internal';
+import { areBtreeSetsEqual, checkTxType } from '~/utils/internal';
 
 /**
  * @hidden
  */
 export type SetAssetStatParams = { asset: FungibleAsset } & SetTransferRestrictionStatParams;
 
+/**
+ * @hidden
+ */
+export interface SetAssetStatsStorage {
+  currentStats: BTreeSet<PolymeshPrimitivesStatisticsStatType>;
+}
+
+/**
+ * @hidden
+ */
+function getStatUpdates(
+  stat:
+    | AddCountStatParams
+    | AddBalanceStatParams
+    | AddClaimCountStatParams
+    | AddClaimBalanceStatParams,
+  context: Context
+): BTreeSet<PolymeshPrimitivesStatisticsStatUpdate> | undefined {
```

Comment:
```suggestion
): BTreeSet<PolymeshPrimitivesStatisticsStatUpdate> {
```

Is there a meaningful difference between an empty set and undefined? Returning an empty BTreeSet instead is cleaner if it can be done.